agent/containers/images/image-test-script.sh

```diff
@@ -60,9 +111,20 @@ wait_keypress 120
 anotherhost=y
 while [ $anotherhost == "y" ]
 do
-    read -p "Hostname: " hostvar
-    echo $hostvar >> ${pbench_run}/remotes.lis
-    read -p "Another host? (y/n)" anotherhost
+    read -p "Hostname (or press 'ENTER' for default '$CONTROLLER'): " hostvar
+    if [ -z "$hostvar" ]
+    then
+        hostvar=$CONTROLLER
+    fi
+    ping_first=$(ping -c 1 -W 1 ${hostvar} 2>&1 | grep 'Name or service not known')
+    ping_second=$(ping -c 1 -W 1 ${hostvar} 2>&1 | grep '100% packet loss')
+    if [ -n "$ping_first" ] || [ -n "$ping_second" ]
+    then
+        printf -- "Host specified either does not exist or is unreachable, please try again:\n"
+    else
+        echo $hostvar >> ${pbench_run}/remotes.lis
```

Comment:
You're appending to the file, here...how do you know that the file was empty to start with?

Having it persist from run to run might be a feature...but, you'll want to display the contents to the user, get confirmation that it's acceptable, and you might want to go down the list and ping them.

Also, see previous comments regarding improving the code in this loop.

Reply:
In the lines above the run directory is wiped and remade, so it should be fresh for each run:
`rm -rf ${pbench_run}`
`mkdir ${pbench_run}`
But yeah the loop needs improvement